lib/index.js

```diff
@@ -540,7 +543,13 @@ api.verify = async ({
   const now = new Date();
   if(type === 'nonce') {
     for(const token of tokenValue) {
-      if(token.sha256 === prefixedHash(hash)) {
+      let combinedHash;
+      if(clientId) {
+        combinedHash = prefixedHash(`${hash}:${clientId}`);
+      } else {
+        combinedHash = prefixedHash(hash);
+      }
+      if(token.sha256 === combinedHash) {
```

Comment:
So I'm also noticing that this code should have been using `crypto.timingSafeEqual` from line 612 below where the verification is performed. I wonder if we could reorg the code a bit so we don't have to do the combined hash and the comparison twice and so we don't accidentally have bugs like this where we do a constant time compare in one place but not the other.

Reply:
Yeah, it's pretty good -- if we can create a local helper for the comparison that would finish up the refactor part nicely.